### ISSUE TITLE
Default basemap is no more working in 2.2

### ISSUE THREAD

**Issue by rbovard** [camptocamp/ngeo#3280]
* GeoMapFish version: 2.2.1

Permalink directly where we can reproduce the issue:
* KO: https://map-demo.nyon.ch/dev/mobile/theme/poi (2.2)
* OK: https://map.nyon.ch/mobile/theme/poi (2.1)

FYI the `default_basemap` for POI theme is `asitvd.fond_couleur`.

**Comment by fredj** [comment]
The `default_basemap` from the theme is not taken into account because there's a `default_basemap` configured for the anonymous user. In the yaml config file under "functionalities > anonymous > default_basemap"

**Comment by rbovard** [comment]
So the behaviour changed since 2.1?
So how can I get the same as before?

**Comment by fredj** [comment]
It was wrong in 2.1: the `default_basemap` from the user configuration was never used.

The order for setting the basemap is:
 * from the permalink
 * from the user config
 * from the theme config
 * the first layer from `background_layers`

**Comment by fredj** [comment]
@rbovard to use the basemap from the theme instead of the user config, you can remove "functionalities > anonymous > default_basemap" from the yaml config file

**Comment by rbovard** [comment]
Thanks @fredj 
I just did it but it doesn't work... (see first permalink)

```

diff --git a/vars_nyon.yaml b/vars_nyon.yaml
index 8ba005d..e264fb3 100644
--- a/vars_nyon.yaml
+++ b/vars_nyon.yaml
@@ -48,7 +48,6 @@ vars:
             - 02 A4 landscape
             - 03 A3 portrait
             - 04 A3 landscape
-            default_basemap: plan
             authorized_plugins: []

             # The name of the theme to use as the default theme.
```

**Comment by fredj** [comment]
There's still an `default_basemap` for the anonymous user:
https://map-demo.nyon.ch/dev/wsgi/loginuser

**Comment by rbovard** [comment]
Because the change was no more effective...

I rebuild it, could you please test again?

**Comment by rbovard** [comment]
@fredj For the record, the returned content with your change is 

```
{"functionalities": {"authorized_plugins": [], "default_basemap": []}}
```

PS: Because this is a dev instance, this change is not present if I'm working on other stuff...

**Comment by rbovard** [comment]
@fredj Any news on this one?

**Comment by fredj** [comment]
#3374 will fix the issue when `default_basemap` is empty (correct fallback to the theme config).

But there's still an issue on page load (I'm working on this)

**Comment by fredj** [comment]
fixed with #3382

**Comment by rbovard** [comment]
Thanks, it's working.
